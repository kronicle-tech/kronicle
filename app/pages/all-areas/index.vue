<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllAreasTabs />

          <b-alert show="10" dismissible variant="info" class="my-3">
            Click an area's name in the table below to view more information about
            that area
          </b-alert>

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
              <tr>
                <th class="area-name">Name</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr v-for="area in areas" :key="area.id">
                <td class="area-name table-primary">
                  <AreaName :area="area" />
                </td>
                <td>
                  <Markdown :markdown="area.description" />
                </td>
              </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BAlert, BCol, BContainer, BRow} from 'bootstrap-vue'
import { Area } from '~/types/kronicle-service'
import AllAreasTabs from '~/components/AllAreasTabs.vue'
import AreaName from '~/components/AreaName.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllAreasTabs,
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    AreaName,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id,name,description)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    return {
      areas,
    }
  },
  data() {
    return {
      areas: [] as Area[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Areas',
    }
  },
})
</script>
