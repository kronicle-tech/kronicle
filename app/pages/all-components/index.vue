<template>
  <div>
    <AllComponentsTabs />
    <b-alert show="10" dismissible variant="info" class="my-3">
      Click a component's name in the table below to view more information about
      that component
    </b-alert>

    <ComponentsView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BAlert } from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component } from '~/types/kronicle-service'
import AllComponentsTabs from '~/components/AllComponentsTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AllComponentsTabs,
    'b-alert': BAlert,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId)`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    store.commit('componentFilters/initialize', {
      components,
      route,
    })

    return {
      components,
    }
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Components',
    }
  },
})
</script>
