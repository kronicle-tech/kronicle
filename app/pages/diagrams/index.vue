<template>
  <div class="m-3">
    <DiagramsView :diagrams="diagrams" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { Diagram } from '~/types/kronicle-service'
import DiagramsView from '~/components/DiagramsView.vue'

export default Vue.extend({
  components: {
    DiagramsView,
  },
  async asyncData({ $config }) {
    const diagrams = await fetch(
      `${$config.serviceBaseUrl}/v1/diagrams?fields=diagrams(id,name,type,tags)`
    )
      .then((res) => res.json())
      .then((json) => json.diagrams)

    return {
      diagrams,
    }
  },
  data() {
    return {
      diagrams: [] as Diagram[],
    }
  },
  head() {
    return {
      title: 'Kronicle - Diagrams',
    }
  },
})
</script>
