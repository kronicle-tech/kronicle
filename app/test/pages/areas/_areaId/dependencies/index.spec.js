import Index from '~/pages/areas/_areaId/dependencies/index.vue'
import { createPageWrapper } from '~/test/pages/pageUtils'
import {
  createArea,
  createComponent,
  createComponentDependencies,
  createSubComponentDependencies,
  createSummaryWithEmptyComponentAndSubComponentDependencies,
} from '~/test/testDataUtils'

describe('Index', () => {
  const route = {
    params: {
      areaId: 'test-area-id-1',
    },
  }
  let area
  let allComponents
  let summary
  let wrapperActions
  let wrapper
  async function createWrapper() {
    wrapper = await createPageWrapper(Index, {
      route,
      serviceRequests: {
        '/v1/areas/test-area-id-1?fields=area(id,name,components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId,states(environmentId,pluginId)))':
          {
            responseBody: { area },
          },
        '/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId)':
          {
            responseBody: { components: allComponents },
          },
        '/v1/summary?fields=summary(componentDependencies,subComponentDependencies)':
          {
            responseBody: { summary },
          },
      },
    })
    for (const wrapperAction of wrapperActions) {
      await wrapperAction(wrapper)
    }
  }

  beforeEach(() => {
    area = createArea({ areaNumber: 1 })
    area.components = [
      createComponent({ componentNumber: 1 }),
      createComponent({ componentNumber: 2 }),
      createComponent({ componentNumber: 3 }),
    ]
    allComponents = [].concat(area.components, [
      createComponent({ componentNumber: 4 }),
      createComponent({ componentNumber: 5 }),
      createComponent({ componentNumber: 6 }),
    ])
    summary = createSummaryWithEmptyComponentAndSubComponentDependencies()
    wrapperActions = []
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  test('has the right page title', async () => {
    await createWrapper()
    expect(wrapper.vm.$metaInfo.title).toBe(
      'Kronicle - Test Area Name 1 Area - Visualizations'
    )
  })

  describe('when Get Summary service endpoint returns no dependencies', () => {
    test('renders the page', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('when Get Summary service endpoint returns an array of multiple component dependencies', () => {
    beforeEach(() => {
      summary.componentDependencies = createComponentDependencies()
      summary.subComponentDependencies = createSubComponentDependencies()
    })

    test('renders the page', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
    })

    describe('when the detailed checkbox is checked', () => {
      beforeEach(() => {
        wrapperActions.push(async (wrapper) => {
          await wrapper.get('#detailed-dependencies').trigger('click')
        })
      })

      test('shows detailed dependencies in graph', async () => {
        await createWrapper()
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
