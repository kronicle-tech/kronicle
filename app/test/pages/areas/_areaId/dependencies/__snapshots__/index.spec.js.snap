// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index when Get Summary service endpoint returns an array of multiple component dependencies renders the page 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h1
          class="text-info my-3"
        >
          Test Area Name 1 Area - Dependencies
        </h1>
         
        <div
          class="my-3"
        >
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1"
            target="_self"
          >
            Info
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/dependencies"
            target="_self"
          >
            Dependencies
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/components"
            target="_self"
          >
            Components
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/cross-functional-requirements"
            target="_self"
          >
            
    Cross Functional Requirements
  
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/tech-debts"
            target="_self"
          >
            Tech Debts
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/tests"
            target="_self"
          >
            Tests
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/errors"
            target="_self"
          >
            Errors
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/openapi-specs"
            target="_self"
          >
            OpenAPI Specs
          </a>
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="graph col"
        >
          <svg
            height="1200"
            id="component-dependency-graph"
            overflow="auto"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="component-dependency-graph-background"
              fill-opacity="0"
            />
             
            <defs>
              <marker
                class="dependency-marker all-dependency-marker"
                id="component-dependency-graph-dependency-marker-all"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker manual-dependency-marker"
                id="component-dependency-graph-dependency-marker-manual"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker scope-related-dependency-marker"
                id="component-dependency-graph-dependency-marker-scope-related"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker scoped-dependency-marker"
                id="component-dependency-graph-dependency-marker-scoped"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker related-dependency-marker"
                id="component-dependency-graph-dependency-marker-related"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker direct-dependency-marker"
                id="component-dependency-graph-dependency-marker-direct"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker selected-dependency-marker"
                id="component-dependency-graph-dependency-marker-selected"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
            </defs>
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
          </svg>
        </div>
         
        <div
          class="col-md-3"
        >
          <transition-stub
            css="true"
            enteractiveclass=""
            enterclass=""
            entertoclass="show"
            leaveactiveclass=""
            leaveclass="show"
            leavetoclass=""
            name=""
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="alert alert-dismissible alert-info"
              role="alert"
            >
              <button
                aria-label="Close"
                class="close"
                type="button"
              >
                Ã—
              </button>
              
          Hover over or click a component's dot in the dependencies diagram to see
          more information about it
        
            </div>
          </transition-stub>
           
          <div
            class="my-3"
          >
            <!---->
             
            <div
              class="form-row form-group my-3 mr-2"
              id="__BVID__66"
              role="group"
            >
              <label
                class="col-6 col-form-label col-form-label-sm"
                for="graph-scope-related-radius"
                id="__BVID__66__BV_label_"
              >
                Radius:
              </label>
              <div
                class="col"
              >
                <select
                  class="custom-select custom-select-sm"
                  id="graph-scope-related-radius"
                >
                  <option
                    value="0"
                  >
                    0
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="form-row form-group my-3 mr-2"
              id="__BVID__68"
              role="group"
            >
              <label
                class="col-6 col-form-label col-form-label-sm"
                for="graph-zoom"
                id="__BVID__68__BV_label_"
              >
                Zoom:
              </label>
              <div
                class="col"
              >
                <select
                  class="custom-select custom-select-sm"
                  id="graph-zoom"
                >
                  <option
                    value="25"
                  >
                    25%
                  </option>
                  <option
                    value="50"
                  >
                    50%
                  </option>
                  <option
                    value="75"
                  >
                    75%
                  </option>
                  <option
                    value="100"
                  >
                    100%
                  </option>
                  <option
                    value="125"
                  >
                    125%
                  </option>
                  <option
                    value="150"
                  >
                    150%
                  </option>
                  <option
                    value="200"
                  >
                    200%
                  </option>
                  <option
                    value="400"
                  >
                    400%
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="custom-control custom-checkbox"
            >
              <input
                class="custom-control-input"
                id="detailed-dependencies"
                type="checkbox"
                value="true"
              />
              <label
                class="custom-control-label"
                for="detailed-dependencies"
              >
                
            Detailed dependencies
          
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Index when Get Summary service endpoint returns an array of multiple component dependencies when the detailed checkbox is checked shows detailed dependencies in graph 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h1
          class="text-info my-3"
        >
          Test Area Name 1 Area - Dependencies
        </h1>
         
        <div
          class="my-3"
        >
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1"
            target="_self"
          >
            Info
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/dependencies"
            target="_self"
          >
            Dependencies
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/components"
            target="_self"
          >
            Components
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/cross-functional-requirements"
            target="_self"
          >
            
    Cross Functional Requirements
  
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/tech-debts"
            target="_self"
          >
            Tech Debts
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/tests"
            target="_self"
          >
            Tests
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/errors"
            target="_self"
          >
            Errors
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/openapi-specs"
            target="_self"
          >
            OpenAPI Specs
          </a>
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="graph col"
        >
          <svg
            height="1200"
            id="component-dependency-graph"
            overflow="auto"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="component-dependency-graph-background"
              fill-opacity="0"
            />
             
            <defs>
              <marker
                class="dependency-marker all-dependency-marker"
                id="component-dependency-graph-dependency-marker-all"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker manual-dependency-marker"
                id="component-dependency-graph-dependency-marker-manual"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker scope-related-dependency-marker"
                id="component-dependency-graph-dependency-marker-scope-related"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker scoped-dependency-marker"
                id="component-dependency-graph-dependency-marker-scoped"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker related-dependency-marker"
                id="component-dependency-graph-dependency-marker-related"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker direct-dependency-marker"
                id="component-dependency-graph-dependency-marker-direct"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker selected-dependency-marker"
                id="component-dependency-graph-dependency-marker-selected"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
            </defs>
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
          </svg>
        </div>
         
        <div
          class="col-md-3"
        >
          <transition-stub
            css="true"
            enteractiveclass=""
            enterclass=""
            entertoclass="show"
            leaveactiveclass=""
            leaveclass="show"
            leavetoclass=""
            name=""
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="alert alert-dismissible alert-info"
              role="alert"
            >
              <button
                aria-label="Close"
                class="close"
                type="button"
              >
                Ã—
              </button>
              
          Hover over or click a component's dot in the dependencies diagram to see
          more information about it
        
            </div>
          </transition-stub>
           
          <div
            class="my-3"
          >
            <!---->
             
            <div
              class="form-row form-group my-3 mr-2"
              id="__BVID__89"
              role="group"
            >
              <label
                class="col-6 col-form-label col-form-label-sm"
                for="graph-scope-related-radius"
                id="__BVID__89__BV_label_"
              >
                Radius:
              </label>
              <div
                class="col"
              >
                <select
                  class="custom-select custom-select-sm"
                  id="graph-scope-related-radius"
                >
                  <option
                    value="0"
                  >
                    0
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="form-row form-group my-3 mr-2"
              id="__BVID__91"
              role="group"
            >
              <label
                class="col-6 col-form-label col-form-label-sm"
                for="graph-zoom"
                id="__BVID__91__BV_label_"
              >
                Zoom:
              </label>
              <div
                class="col"
              >
                <select
                  class="custom-select custom-select-sm"
                  id="graph-zoom"
                >
                  <option
                    value="25"
                  >
                    25%
                  </option>
                  <option
                    value="50"
                  >
                    50%
                  </option>
                  <option
                    value="75"
                  >
                    75%
                  </option>
                  <option
                    value="100"
                  >
                    100%
                  </option>
                  <option
                    value="125"
                  >
                    125%
                  </option>
                  <option
                    value="150"
                  >
                    150%
                  </option>
                  <option
                    value="200"
                  >
                    200%
                  </option>
                  <option
                    value="400"
                  >
                    400%
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="custom-control custom-checkbox"
            >
              <input
                class="custom-control-input"
                id="detailed-dependencies"
                type="checkbox"
                value="true"
              />
              <label
                class="custom-control-label"
                for="detailed-dependencies"
              >
                
            Detailed dependencies
          
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Index when Get Summary service endpoint returns no dependencies renders the page 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h1
          class="text-info my-3"
        >
          Test Area Name 1 Area - Dependencies
        </h1>
         
        <div
          class="my-3"
        >
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1"
            target="_self"
          >
            Info
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/dependencies"
            target="_self"
          >
            Dependencies
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/components"
            target="_self"
          >
            Components
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/cross-functional-requirements"
            target="_self"
          >
            
    Cross Functional Requirements
  
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/tech-debts"
            target="_self"
          >
            Tech Debts
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/tests"
            target="_self"
          >
            Tests
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/errors"
            target="_self"
          >
            Errors
          </a>
           
          <a
            class="btn my-1 btn-secondary"
            href="/areas/test-area-id-1/openapi-specs"
            target="_self"
          >
            OpenAPI Specs
          </a>
        </div>
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="graph col"
        >
          <svg
            height="1200"
            id="component-dependency-graph"
            overflow="auto"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              class="component-dependency-graph-background"
              fill-opacity="0"
            />
             
            <defs>
              <marker
                class="dependency-marker all-dependency-marker"
                id="component-dependency-graph-dependency-marker-all"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker manual-dependency-marker"
                id="component-dependency-graph-dependency-marker-manual"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker scope-related-dependency-marker"
                id="component-dependency-graph-dependency-marker-scope-related"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker scoped-dependency-marker"
                id="component-dependency-graph-dependency-marker-scoped"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker related-dependency-marker"
                id="component-dependency-graph-dependency-marker-related"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker direct-dependency-marker"
                id="component-dependency-graph-dependency-marker-direct"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
              <marker
                class="dependency-marker selected-dependency-marker"
                id="component-dependency-graph-dependency-marker-selected"
                markerHeight="11.5"
                markerUnits="userSpaceOnUse"
                markerWidth="7.1"
                orient="auto-start-reverse"
                refX="10"
                refY="5.75"
                viewBox="0 0 7.1 11.5"
              >
                <path
                  d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
                />
              </marker>
            </defs>
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
             
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
            <g />
          </svg>
        </div>
         
        <div
          class="col-md-3"
        >
          <transition-stub
            css="true"
            enteractiveclass=""
            enterclass=""
            entertoclass="show"
            leaveactiveclass=""
            leaveclass="show"
            leavetoclass=""
            name=""
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="alert alert-dismissible alert-info"
              role="alert"
            >
              <button
                aria-label="Close"
                class="close"
                type="button"
              >
                Ã—
              </button>
              
          Hover over or click a component's dot in the dependencies diagram to see
          more information about it
        
            </div>
          </transition-stub>
           
          <div
            class="my-3"
          >
            <!---->
             
            <div
              class="form-row form-group my-3 mr-2"
              id="__BVID__43"
              role="group"
            >
              <label
                class="col-6 col-form-label col-form-label-sm"
                for="graph-scope-related-radius"
                id="__BVID__43__BV_label_"
              >
                Radius:
              </label>
              <div
                class="col"
              >
                <select
                  class="custom-select custom-select-sm"
                  id="graph-scope-related-radius"
                >
                  <option
                    value="0"
                  >
                    0
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="form-row form-group my-3 mr-2"
              id="__BVID__45"
              role="group"
            >
              <label
                class="col-6 col-form-label col-form-label-sm"
                for="graph-zoom"
                id="__BVID__45__BV_label_"
              >
                Zoom:
              </label>
              <div
                class="col"
              >
                <select
                  class="custom-select custom-select-sm"
                  id="graph-zoom"
                >
                  <option
                    value="25"
                  >
                    25%
                  </option>
                  <option
                    value="50"
                  >
                    50%
                  </option>
                  <option
                    value="75"
                  >
                    75%
                  </option>
                  <option
                    value="100"
                  >
                    100%
                  </option>
                  <option
                    value="125"
                  >
                    125%
                  </option>
                  <option
                    value="150"
                  >
                    150%
                  </option>
                  <option
                    value="200"
                  >
                    200%
                  </option>
                  <option
                    value="400"
                  >
                    400%
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="custom-control custom-checkbox"
            >
              <input
                class="custom-control-input"
                id="detailed-dependencies"
                type="checkbox"
                value="true"
              />
              <label
                class="custom-control-label"
                for="detailed-dependencies"
              >
                
            Detailed dependencies
          
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
